app/controllers/smart_proxies_controller.rb

```diff
@@ -63,8 +66,12 @@ def destroy
 
   private
 
-  def requested_data(data_name)
-    data = @smart_proxy.public_send(data_name)
+  def find_status
+    @proxy_status = @smart_proxy.proxy_status
+  end
+
+  def requested_data(object, method)
+    data = object.public_send(method)
     render :json => {:success => true, :message => data}
```

Comment:
I was going to suggest this exact same thing on the original PR but didn't want to delay it any more. If you could do it it'd be great. Basically it'd be similar to the way `fetch_proxy_data` works. Your external calls would look like

```
requested_data do 
  @proxy_status[:tftp].server
end
```

I'd also suggest changing the name `requested_data` (generic and meaningless) to something more specific, even `fetch_proxy_data` again if you want?